src/pages/index.js

```diff
@@ -0,0 +1,330 @@
+
+// Form Variables
+const editProfileForm = document.forms["edit-profile-form"];
+const newPostForm = document.forms["new-post-form"];
+const avatarForm = document.forms["avatar-form"];
+const editProfileName = editProfileForm.querySelector("#name");
+const editProfileDesc = editProfileForm.querySelector("#description");
+const newPostLink = newPostForm.querySelector("#post-link");
+const newPostCapt = newPostForm.querySelector("#post-caption");
+const avatarLink = avatarForm.querySelector("#avatar-link");
+const deletePostForm = document.forms["delete-post-form"];
+
+// Close buttons
+const closeButtons = document.querySelectorAll(".modal__close-button");
+
+// ---------------------------------------------------------------------
+const validatorProfile = new FormValidator(settings, editProfileForm);
+const validatorNewPost = new FormValidator(settings, newPostForm);
+const validatorAvatar = new FormValidator(settings, avatarForm);
+// ---------------------------------------------------------------------
+
+// API
+const api = new Api({
+    url: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+        authorization: "13430c32-a8ca-4f0e-9fc3-4da36101b73c",
+        "Content-Type": "application/json",
+    },
+});
+
+// Modal functions
+
+function toggleModal(modal) {
+    modal.classList.toggle("modal_opened");
+    if (modal.classList.contains("modal_opened")) {
+        enableEscapeClose();
+    } else {
+        disableEscapeClose();
+    }
+}
+
+function handleOpenModalProfile() {
+    editProfileName.value = profileName.textContent;
+    editProfileDesc.value = profileDesc.textContent;
+    editProfileName.dispatchEvent(new Event("input"));
+    editProfileDesc.dispatchEvent(new Event("input"));
+    toggleModal(editProfileModal);
+}
+
+function handleOpenModalNewPost() {
+    toggleModal(newPostModal);
+}
+
+function handleOpenModalAvatar() {
+    toggleModal(avatarModal);
+}
+
+function handleProfileFormSubmit(evt) {
+    evt.preventDefault();
+    loading(true, editProfileForm.querySelector(".modal__submit-button"));
+
+    api.editUserInfo({
+        name: editProfileName.value,
+        about: editProfileDesc.value,
+    })
+        .then((res) => {
+            profileName.textContent = res.name;
+            profileDesc.textContent = res.about;
+        })
+        .catch((err) => {
+            console.error(err);
+        })
+        .finally(() => {
+            loading(
+                false,
+                editProfileForm.querySelector(".modal__submit-button")
+            );
+        });
+
+    toggleModal(editProfileModal);
+}
+
+function handleAvatarFormSubmit(evt) {
+    evt.preventDefault();
+    loading(true, avatarForm.querySelector(".modal__submit-button"));
+    api.editAvatar({ avatar: avatarLink.value })
+        .then((res) => {
+            profileAvatar.src = res.avatar;
+            profileAvatar.alt = res.name;
+            avatarForm.reset();
```

Comment:
* Please, **disable** the `avatar` submit button as well after submission https://snipboard.io/FfeV2J.jpg 


```js
            validatorAvatar.resetValidation();


```